docs/core/deploying/trimming/prepare-libraries-for-trimming.md

```diff
@@ -8,360 +8,276 @@ ms.date: 06/12/2023
 
-static void InitializeTypeField()
-{
-    MyLibrary.type = typeof(System.Tuple);
-}
-```
+:::code language="csharp" source="~/docs/core/deploying/trimming/snippets/MyLibrary/Class3.cs" id="snippet_UMH2" highlight="1":::
 
 In this case, the trim analysis keeps public methods of <xref:System.Tuple>, and produces further warnings.
 
 ## Recommendations
 
-In general, try to avoid reflection if possible. When using reflection, limit it in scope so that it's reachable only from a small part of the library.
-
-- Avoid using non-understood patterns in places like static constructors that result in the warning propagating to all members of the class.
-- Avoid annotating virtual methods or interface methods, which will require all overrides to have matching annotations.
-- In some cases, you're able to mechanically propagate warnings through your code without issues. Sometimes this results in much of your public API being annotated with `RequiresUnreferencedCode`, which is the right thing to do if the library indeed behaves in ways that can't be understood statically by the trim analysis.
-- In other cases, you might discover that your code uses patterns that can't be expressed in terms of the `DynamicallyAccessedMembers` attributes, even if it only uses reflection to operate on statically known types. In these cases, you may need to reorganize some of your code to make it follow an analyzable pattern.
-- Sometimes the existing design of an API renders it mostly trim-incompatible, and you may need to find other ways to accomplish what it's doing. A common example is reflection-based serializers. In these cases, consider adopting other technology like source generators to produce code that is more easily statically analyzed.
+* ***Avoid*** reflection when possible. When using reflection, minimize reflection scope so that it's reachable only from a small part of the library.
+* Annotate code with `DynamicallyAccessedMembers` to statically express the trimming requirements statically when possible.
+* Consider reorganizing code to make it follow an analyzable pattern that can be annotated with `DynamicallyAccessedMembers`
+* When code is incompatible with trimming, annotate it with `RequiresUnreferencedCode` and propagate this annotation to callers until the relevant public APIs are annotated.
+* Avoid using code that uses reflection in a way not understood by the static analysis. For example, reflection in static constructors should be avoided. Using reflection in static constructors result in the warning propagating to all members of the class.
```

Comment:
```suggestion
* Avoid using code that uses reflection in a way not understood by the static analysis. For example, reflection in static constructors should be avoided. Using statically unanalyzable reflection in static constructors result in the warning propagating to all members of the class.
```

It's OK to use reflection which the analysis can figure out - even in a static constructor (in fact there are many use cases for this and we do that in several places in our libraries as well), the problem is trim incompatible code in static constructor - which is hard to reason about who to blame.